### PR TITLE
Avoid setting hostname-override

### LINKED ISSUE #10590: Possibility to disable hostname-override in kubelet
According to the CIS benchmark kubelet shouldn't have the hostname-override flag. So it will be great to have the possibility to disable hostname-override.

Ref #6150

### PR DESCRIPTION
**Pull request by olemarkus**
This is an alternative to #10595

Fixes #10590

### CODE REVIEW

File: nodeup/pkg/model/context.go [hunk 1]
```diff
@@ -566,42 +566,33 @@ func (c *NodeupModelContext) BuildLegacyPrivateKeyTask(ctx *fi.ModelBuilderConte
 
 // NodeName returns the name of the local Node, as it will be created in k8s
 func (c *NodeupModelContext) NodeName() (string, error) {
-	// This mirrors nodeutil.GetHostName
+
+	// If user explicitly sets hostnameOverride, we respect that even though the CCM will likely just ignore this.
```
Comment (hakman) [inline nodeup/pkg/model/context.go:570]: Maybe we should add validation ad just remove the option?

**Comment by olemarkus** [comment]
As we eventually will move to external CCMs we do need to maintain hostname overrides to match what CCM expects. This goes for both kube-proxy and kubelet.

/hold

**Pull request closed without merging**